### ISSUE TITLE
iotools.create_array fails when the data file is very big

### ISSUE THREAD

**Issue by rdarie** [tridesclous/tridesclous#112]
Hi,

I think I came across a subtle issue with this check here:
https://github.com/tridesclous/tridesclous/blob/192dc19c84aad4fcfea9af489cefa921469aec9b/tridesclous/iotools.py#L129-L130
For a large enough array, the product of ints overflows and fails to pass the check:
```
DataIO <id: 2076988629624>
workdir: ..\201901271000-Proprio\tdc_Trial003
sample_rate: 30000.0
total_channel: 112
channel_groups: 0 (96ch), 1 (16ch)
nb_segment: 1
length: 31987278
durations: 1066.2 s.
c:\users\radu\documents\github\tridesclous\tridesclous\iotools.py(130)create_array()
-> if np.prod(shape)>0:
(Pdb) shape
(31987278, 96)
(Pdb) np.prod(shape)
-1224188608
#  file is ~1000 sec long, sampled at 30kHz
(Pdb) 31987278 / 3e4
1066.2426
(Pdb) np.prod(shape, dtype=np.int64)
3070778688
```
That's on Windows. I don't get this issue on Linux with the same file, so I imagine it's a weird problem with the underlying choice of default int. Casting to a bigger int seems to fix it.
Also, I am new to tdc (great resource, by the way!) so I wonder if it's normal for it to be allocating the entire array size at this step. I was only requesting a short 24 second snippet in the preprocessor.

**Comment by samuelgarcia** [comment]
Oups.
Do you have python 32bit on windows ?
on python.org python 32bit is download by default.

I will try to fix it.

The entire array is allocated by with memmap so it do not cost almost nothing.
It avoid to allocate the same array again and again when changing params for filetring.

**Comment by samuelgarcia** [comment]
Many the easiest fix is to do 
``` if np.prod(shape)!=0: ```

**Comment by rdarie** [comment]
Thanks for the quick reply! Turns out this also affects 
https://github.com/tridesclous/tridesclous/blob/192dc19c84aad4fcfea9af489cefa921469aec9b/tridesclous/iotools.py#L245-L247
I learned a lot more today about numpy int types than I thought possible, haha! I think the issue is that a c++ long is 4 bytes on Windows and 8 bytes on Unix, regardless of 32bit or 64bit (I'm using 64-bit 3.6.5). np.prod returns type np.int_, which is just an alias for the platform's long.
